net/haproxy/src/opnsense/service/templates/OPNsense/HAProxy/haproxy.conf

```diff
@@ -459,8 +460,44 @@
 {%             else %}
 {%               set defaultbackend_option = '' %}
 {%             endif %}
+{#             # Store match function from configured match type #}
+{%             set backend_file_matchtype = action_data.map_use_backend_file_matchtype|default("dom") %}
+{%             if backend_file_matchtype in ("str", "dom", "reg", "beg", "end", "sub") %}
+{%               set backend_file_matchtype_func = "map_" ~ backend_file_matchtype %}
+{%             else %}
+{%               set backend_file_matchtype_func = "map_dom" %}
+{%             endif %}
 {#             # Finally add map file to config #}
-{%             do action_options.append('use_backend %[req.hdr(host),lower,map_dom(' ~ mapfile_path ~ defaultbackend_option ~ ')]') %}
+{%             set backend_file_hostname_snisupport = action_data.map_use_backend_file_snisupport|default("hostonly") %}
+{%             if backend_file_hostname_snisupport in ("sniandhost", "snionly") %}
+{%               set backend_file_global_options = ['# NOTE: Map files SNI support', 'tcp-request inspect-delay 200ms', 'tcp-request content accept if { req.ssl_hello_type 1 }', ''] %}
```

Comment:
Do you have an usecase where this is required? I was led by the impression that SNI comes with the very first packet, thus any delay would indicate a non-SNI connection anyway.
PS: the 500 bytes capture buffer chosen above is below the minimum MTU required by IP/IPv6 for the same reason.

Reply:
Using hardcoded values is always something we should try to avoid. It may work for one use-case, but fail in other scenarios.

However, I don't think this option should silently be added here. Instead it should be documented that this feature relies on it and that the user needs to add it (because it's already available in os-haproxy with a customizable inspection delay).